recipes/views.py

```diff
@@ -0,0 +1,63 @@
+from django.urls import reverse_lazy
+from django.views.generic import (CreateView, DetailView, DeleteView,
+                                  ListView, UpdateView)
+
+from recipes.models import Recipe
+
+
+class RecipeCreateView(CreateView):
+    """
+
+    View to create a recipe
+    """
+    model = Recipe
+    fields = ['user', 'title', 'description', 'servings', 'ingredients', 'preparation_time',
+              'cook_time', 'difficulty']
+
+    def get_success_url(self):
+        return reverse_lazy('detail-single-recipe', kwargs={'pk': self.object.id})
+
+
+class RecipeListView(ListView):
+    """
+
+    View that lists all recipes OR filters by Ingredient
+    """
+    model = Recipe
+    context_object_name = 'recipe_list'
```

Comment:
`context_object_name` is used here in the `ListView`


```
    def get_context_object_name(self, object_list):
        """Get the name of the item to be used in the context."""
        if self.context_object_name:
            return self.context_object_name
        elif hasattr(object_list, 'model'):
            return '%s_list' % object_list.model._meta.model_name
        else:
            return None
```
So by default, every view gets a generic `object` in context, and a name context variable based on the model name

ie `object_list` and `recipe_list`, so you don't need to set one yourself if it's already the same as the automatic one